src/preconditions/ClientPermissions.ts

```diff
@@ -30,14 +42,46 @@ export class CorePrecondition extends Precondition {
 
 		const permissions = message.guild ? channel.permissionsFor(message.client.id) : this.dmChannelPermissions;
 
-		if (!permissions) {
+		return this.sharedRun(required, permissions, 'message');
+	}
+
+	public chatInputRun(interaction: CommandInteraction, _: Command, context: UserPermissionsPreconditionContext) {
+		const required = context.permissions ?? new Permissions();
+		const permissions = interaction.inCachedGuild()
+			? // We are in a cached guild, try to get our permissions
+			  (interaction.channel as GuildTextBasedChannel).permissionsFor(interaction.applicationId)
+			: // We are in an uncached guild so we don't know our permissions
+			// TODO(vladfrangu): Do we want to fetch the channel somehow and use that?
```

Comment:
Do we? @kyranet @vladfrangu. I cannot answer this without some context. Write up some pro's and cons for doing so for starters.

Reply:
As I said before, my experience with interaction commands is basically null. I'll leave that answer to the experts.

Reply:
@kyranet this isn't related to interaction commands per se. This is the `ClientPermissions` precondition and it's related to how we handle a cached  (or non-cached) channel.

Reply:
Sure, let's fetch, better safe than sorry, and not like users can really pre-fetch before that happens.